### DISCUSSION TITLE
Create new hosts or services via REST API

### DISCUSSION THREAD

**Discussion by Beatlor** [sni/thruk#1145] [Q&A] [answered]
Hello Thruk users,

for an automation and lifecycle process I want to create and delete hosts and services  inside of thruk via the REST API.

The scripts are all based on curl and JSON. I can't find a way to create a host via the rest api. Tested already so many ways include the
header authentication which looks fine. 

I already can get all data reading out of Thruk but the other way - no chance.

Has anybody already created a new host/service via curl and JSON? So I have no more ideas.

curl --insecure -s -S -X POST -H "Content-Type: application/json" -H "X-Thruk-Auth-User:omdadmin" -H "X-Thruk-Auth-Key:SECRETTOKEN" --data '{"host_name":"TESTHOST01"}' https:/host.domain.com/site/thruk/r/hosts/TESTHOST01/config

I'm very very happy for a good idea!

**Comment by sni** [comment]
You need to POST to /config/objects as described here: https://thruk.org/documentation/rest.html#_post-config-objects

**Reply by Beatlor** [reply]
Thx for your support but:

Always getting an error

Command with curl $MONHOST/config/objects?:TYPE=host&:FILE=testhosts.cfg

 line 62: :FILE=testhosts.cfg: command not found

{
   "code" : 400,
   "help" : "Create new object. Besides the actual object config, requires\n2 special paramters :FILE and :TYPE.",
   "message" : ":FILE is a required parameter."
}

If I remove the :TYPE I get the error the :TYPE is a required parameter.

**Comment by sni** [comment, marked as answer]
when using & in the shell, you need to put it in quotes, otherwise the shell will interpret that as a background command.
try something like this:

```
curl \
  -X POST \
  --header "Content-Type: application/json" \
  -d '{":TYPE": "host", ":FILE":"testhosts.cfg","name":"testhost123"}' \
  http://localhost:3000/thruk/r/config/objects
```

**Reply by Beatlor** [reply]
That's was the issue. Thx for your support.
